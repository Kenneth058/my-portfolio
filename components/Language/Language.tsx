import * as React from 'react'
import Link from 'next/link'

interface Props {
  href: string
  locale: string
}

const Language: React.FunctionComponent<Props> = ({ href, locale, children }) => (
  <Link href={href} locale={locale}>
    <a className="mx-2.5 inline-block transition-all duration-300 transform hover:-translate-y-2.5">
      {children}
    </a>
  </Link>
)

export default Language
