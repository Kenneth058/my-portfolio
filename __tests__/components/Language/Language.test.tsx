import * as React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Language from '../../../components/Language/Language'

describe('Language Component', () => {
  it('renders', () => {
    render(
      <Language href="/" locale="es">
        ğŸ‡¨ğŸ‡·
      </Language>
    )

    const text = screen.getByText(/ğŸ‡¨ğŸ‡·/i)
    expect(text).toBeInTheDocument()
  })

  it('snapshot', () => {
    const tree = renderer
      .create(
        <Language href="/" locale="es">
          ğŸ‡¨ğŸ‡·
        </Language>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
