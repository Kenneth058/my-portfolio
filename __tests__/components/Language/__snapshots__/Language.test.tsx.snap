// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Component snapshot 1`] = `
<a
  className="mx-2.5 inline-block transition-all duration-300 transform hover:-translate-y-2.5"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  ðŸ‡¨ðŸ‡·
</a>
`;
